test/ModifyLiquidity.t.sol

```diff
@@ -0,0 +1,139 @@
+        );
+        assertGt(updated.liquidity, position.liquidity);
+    }
+
+    function test_modifyLiquidity_samePosition_withSalt_isUpdated() public {
+        Position.Info memory position = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+        assertEq(position.liquidity, 0);
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_SALT, ZERO_BYTES, false, false);
+        position = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+        assertGt(position.liquidity, 0);
+
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_SALT, ZERO_BYTES, false, false);
+        Position.Info memory updated = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+        assertGt(updated.liquidity, position.liquidity);
+    }
+
+    function test_modifyLiquidity_sameTicks_withDifferentSalt_isNotUpdated() public {
+        Position.Info memory positionNoSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_NO_SALT.tickLower, LIQ_PARAM_NO_SALT.tickUpper, 0
+        );
+
+        Position.Info memory positionSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+        assertEq(positionNoSalt.liquidity, 0);
+        assertEq(positionSalt.liquidity, 0);
+
+        // Modify the liquidity with the salt.
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_SALT, ZERO_BYTES, false, false);
+
+        positionNoSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_NO_SALT.tickLower, LIQ_PARAM_NO_SALT.tickUpper, 0
+        );
+
+        positionSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+
+        assertEq(positionNoSalt.liquidity, 0); // This position does not have liquidity.
+        assertGt(positionSalt.liquidity, 0); // This position does.
+
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_NO_SALT, ZERO_BYTES, false, false); // Now the positions should have the same liquidity.
+
+        positionNoSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_NO_SALT.tickLower, LIQ_PARAM_NO_SALT.tickUpper, 0
+        );
+
+        positionSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+
+        assertEq(positionNoSalt.liquidity, positionSalt.liquidity);
```

Comment:
its already implied, but for clarity we should also assert that `positionSalt` was _not_ modifed because we added liquidity to `postionNoSalt`